go/sql/types.go

```diff
@@ -44,6 +50,18 @@ func (this *Column) convertArg(arg interface{}) interface{} {
 		if encoding, ok := charsetEncodingMap[this.Charset]; ok {
 			arg, _ = encoding.NewDecoder().String(s)
 		}
+
+		if this.Type == BinaryColumnType {
+			size := len([]byte(arg.(string)))
+			if uint(size) < this.BinaryOctetLength {
+				buf := bytes.NewBuffer([]byte(arg.(string)))
+				for i := uint(0); i < (this.BinaryOctetLength - uint(size)); i++ {
+					buf.Write([]byte{0})
+				}
+				arg = buf.String()
+			}
+		}
+
```

Comment:
Could you please explain what the logic above does? Thank you!

Reply:
This is all about this issue https://github.com/github/gh-ost/issues/909
Also, U can read this https://dev.mysql.com/doc/refman/5.7/en/binary-varbinary.html
```
You should consider the preceding padding and stripping 
characteristics carefully if you plan to use the BINARY 
data type for storing binary data and you require that 
the value retrieved be exactly the same as the value stored. 
The following example illustrates how 0x00-padding 
of BINARY values affects column value comparisons
```
Please, read @cenkore's first message:
```
The reason is that the last byte of 0x00 was truncated 
when mysql wrote Binlog, so the data obtained when 
gh-ost parsed binlog through go-mysql was already truncated
```
So, this part of code appends missing zeroes.

We use this fork on production about a month, and all is OK.

Reply:
Thank you. Is there a limit on how many zeros you need to pad with, or can this number be very high?

Reply:
@shlomi-noach Filling a high number zero bytes may affect processing efficiency. after thinking about it, we can fix it more lighter as below, other scenarios will not cause data consistency problems. https://github.com/github/gh-ost/pull/915/commits/40bc5aedf2ff1d85c10c4d31a594d0b176d149d7
- only handle the binary column in the unique key
- only handle update/delete dml event